src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingBarracks.java

```diff
@@ -96,6 +98,18 @@ public void onDestroyed()
         super.onDestroyed();
     }
 
+    @Override
+    public void requestUpgrade(final PlayerEntity player, final BlockPos builder)
+    {
+        final UnlockResearchEffect effect = colony.getResearchEffects().getEffect("Barracks", UnlockResearchEffect.class);
```

Comment:
Why is this an effect?!?!?!?
This is a research condition.

Conditions -> Research -> Effects....

Reply:
Oh no hold on i get it!
This is an effect that unlocks a given option.....

Well okey disregard the previous comment! -> But still this needs to be better named. `UnlockBuildingEffect` or `UnlockAbilityEffect` are more appropriately named for buildings and citizen feature locking. Cause right now i had to read 6 classes to understand what is going on.